### DISCUSSION TITLE
Remote Docs

### DISCUSSION THREAD

**Discussion by ghost** [fuma-nama/fumadocs#546] [Docs Feedback]
Can you make an example repo for remote docs as well ? 
I have read the documentation but still can't figure it out. An example repo would be very beneficial.

[Fumadocs UI Template](https://github.com/fuma-nama/fumadocs-ui-template) This one is for local mdx files only.

**Comment by fuma-nama** [comment]
The Remote MDX package has an example under `./examples/mdx-remote`. The docs isn't finished so I can't provide help until the docs for MDX Remote is ready

**Reply by fuma-nama** [reply]
If you don't want to use MDX Remote, use your own implementation by:
- In `page.tsx`, fetch the corresponding page according to the `slug` property in `params` prop. Then, pass your table of contents (toc) to `DocsPage` component, and render the content (for MDX, you can use `next-mdx-remote`).
- In `layout.tsx`, hardcode or generate the page tree, and pass it to `DocsLayout`.
- Configure the search solution, it is more complicated. I would recommend Algolia Search on this, you may also disable search functionality to reduce the complexity

**Reply by ghost** [reply]
Is remote fetching from GitHub repository only possible through mdx-remote for mdx pages?

What about content layer?

**Reply by fuma-nama** [reply]
Contentlayer has its own adapter, you should refer to the docs

**Reply by fuma-nama** [reply]
> Is remote fetching from GitHub repository only possible through mdx-remote for mdx pages

It's a helper to implement that

**Reply by ghost** [reply]
That's what I requested about. Can you make a template repository using content layer?
fetching remote mdx files from GitHub repository?

**Reply by ghost** [reply]
Additionally, could you add an expandable code button? It is helpful for viewing code on mobile devices, as long codes often require extensive swiping. This feature is available in `nextra` as well.

**Reply by fuma-nama** [reply]
I don't think it's useful, it doesn't make the code blocks more readable since you can easily scroll the code block on touch devices, and a good design shouldn't let the users to opt-in, but gives them the best looking by default. You can use CSS to style the pre element in code blocks.
